zap/src/test/java/org/zaproxy/zap/model/ScanEventPublisherUnitTest.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2023 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.model;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.zaproxy.zap.ZAP;
+import org.zaproxy.zap.eventBus.Event;
+import org.zaproxy.zap.eventBus.EventConsumer;
+import org.zaproxy.zap.users.User;
+
+/** Unit test for {@link ScanEventPublisher}. */
+class ScanEventPublisherUnitTest {
+
+    private ScanEventPublisher pub;
+    private TestConsumer sub;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        pub = new ScanEventPublisher() {};
+        sub = new TestConsumer();
+        ZAP.getEventBus().registerPublisher(pub, ScanEventPublisher.getEvents());
+        ZAP.getEventBus().registerConsumer(sub, ScanEventPublisher.class.getCanonicalName());
+    }
+
+    @AfterEach
+    void cleanUp() throws Exception {
+        ZAP.getEventBus().unregisterPublisher(pub);
+        ZAP.getEventBus().unregisterConsumer(sub);
+    }
+
+    @Test
+    void shouldGetPublisherName() {
+        // Given / When
+        String name = pub.getPublisherName();
+
+        // Then
+        assertThat(name, is("org.zaproxy.zap.model.ScanEventPublisher"));
+    }
+
+    void assertIsBasicEvent() {
```

Comment:
This could allow to pass the event type and target, would reduce duplication in one more test.